src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int capacity;
+    private Object[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = new Object[DEFAULT_SIZE];
+        this.capacity = DEFAULT_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
```

Comment:
I would not do this, system array copy is not such a light operation, and adding an element to the end of our storage manually is easier in our case